ethereum/core/src/main/java/org/hyperledger/besu/ethereum/vm/operations/SignExtendOperation.java

```diff
@@ -33,12 +35,30 @@ public Gas cost(final MessageFrame frame) {
 
   @Override
   public void execute(final MessageFrame frame) {
-    final UInt256 value0 = frame.popStackItem().asUInt256();
-    final UInt256 value1 = frame.popStackItem().asUInt256();
+    final UInt256 value0 = UInt256.fromBytes(frame.popStackItem());
+    final UInt256 value1 = UInt256.fromBytes(frame.popStackItem());
 
     // Stack items are reversed for the SIGNEXTEND operation.
-    final UInt256 result = value1.signExtent(value0).asUnsigned();
+    final UInt256 result = signExtend(value1, value0);
+
+    frame.pushStackItem(result.toBytes());
+  }
+
+  private static UInt256 signExtend(final UInt256 v1, final UInt256 v2) {
```

Comment:
Is it the only place we need this method ?

Reply:
Yes.